projects/angular/src/forms/datepicker/providers/datepicker-helper.service.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016-2024 Broadcom. All Rights Reserved.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from '@angular/core';
+
+import { ClrPopoverToggleService } from '../../../utils/popover/providers/popover-toggle.service';
+import { DayModel } from '../model/day.model';
+import { DateFormControlService } from './date-form-control.service';
+import { DateNavigationService } from './date-navigation.service';
+
+@Injectable()
+export class DatePickerHelperService {
+  constructor(
+    private _dateNavigationService: DateNavigationService,
+    private _toggleService: ClrPopoverToggleService,
+    private _dateFormControlService: DateFormControlService
+  ) {}
+
+  selectDay(day: DayModel, emitEvent = !this._dateNavigationService.hasActionButtons): void {
```

Comment:
```suggestion
  selectDay(day: DayModel): void {
```

This can be cleaned up. The `emitEvent` parameter is not needed for this method or for the `notifySelectedDayChanged` method. We can check `hasActionButtons` directly.

See 647b03b91786c2551764dad617a55eeb80c5d888. Feel free to cherry-pick.

Reply:
This can still be cleaned up. Feel free to cherry-pick b7614b3351403cd1b18059979fc7c5631f0e0ec3.

Reply:
This file is no longer required have deleted it.

Reply:
My new suggestion was in a different file for the same thing. But it broke it, so that won't work anyway. Resolved.